client/components/domains/domain-search-results/index.jsx

```diff
@@ -251,9 +251,7 @@ class DomainSearchResults extends React.Component {
 						domainsWithPlansOnly={ this.props.domainsWithPlansOnly }
 						railcarId={ `${ this.props.railcarSeed }-registration-suggestion-${ i + 2 }` }
 						uiPosition={ i + 2 }
-						fetchAlgo={
-							endsWith( suggestion.domain_name, '.wordpress.com' ) ? 'wpcom' : this.props.fetchAlgo
-						}
+						fetchAlgo={ suggestion.fetch_algo ? suggestion.fetch_algo : this.props.fetchAlgo }
```

Comment:
We'll want to make these same changes to `<FeaturedDomainSuggestions />`. I doubt it'll have a significant impact since featured domains would usually default back to `this.props.fetchAlgo`?